src/librustdoc/html/static/js/search.js

```diff
@@ -131,35 +365,43 @@ window.initSearch = function(rawSearchIndex) {
      * @param  {[type]} filterCrates [Crate to search in if defined]
      * @return {[type]}              [A search index of results]
      */
-    function execQuery(query, searchWords, filterCrates) {
-        function itemTypeFromName(typename) {
-            for (var i = 0, len = itemTypes.length; i < len; ++i) {
-                if (itemTypes[i] === typename) {
-                    return i;
-                }
+    function execQuery(queryInfo, searchWords, filterCrates) {
+        // The goal of this new way of storing results is to prevent having too much elements to
+        // sort.
+        function createResults() {
+            var res = {
+                "totalLev": 0,
+                "nb": 0,
+                "elems": [],
+            };
+            for (var i = 0; i <= MAX_LEV_DISTANCE; ++i) {
+                res.elems.push({});
```

Comment:
```suggestion
                res.elems.push(Object.create(null));
```

We don’t want weird misbehavior when someone’s Rust project happens to contain a method called `constructor` or `hasOwnProperty`.

Reply:
Oh wow, I didn't think about that. Thanks!